modal/mount.py

```diff
@@ -23,11 +23,12 @@
 from ._utils.async_utils import aclosing, async_map, synchronize_api
```

Comment:
Do we need to make any changes to `Mount` if we're planning to deprecate it as public API? I could see the argument that it would be confusing to have an inversion between the public interface / internal implementation I guess.

Reply:
Yeah I think the inversion is annoying but also since the `ignore` callable works on `Path` objects and the `condition` we have for mount methods only allows `str` you would basically have to wrap every `ignore` to work on strings instead

Reply:
As per @freider's comment below I opted to add a private method to the `Mount` object for the image methods to use instead and leave the signature of the `Mount` copy & add methods as is